src/BuiltInTools/dotnet-watch/HotReloadDotNetWatcher.cs

```diff
@@ -455,6 +451,11 @@ async Task<ImmutableList<ChangedFile>> CaptureChangedFilesSnapshot(ImmutableDict
                 {
                     // start next iteration unless shutdown is requested
                 }
+                catch (Exception) when ((waitForFileChangeBeforeRestarting = false) == true)
```

Comment:
Why don't you do the `when` in the catch case instead?

```csharp
catch
{
    waitForFileChangeBeforeRestarting = false;
}
```

Reply:
The exception shouldn't be caught.